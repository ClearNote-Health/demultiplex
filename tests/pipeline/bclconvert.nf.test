nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "bclconvert"
    tag "pipeline"

    test("BCL-CONVERT") {

        when {
            params {
                // Limit resources so that this can run on GitHub Actions
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'

                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 5 },
                { assert snapshot(
                        path("$outputDir/multiqc/multiqc_data/bclconvert_lane_counts.txt"),
                        path("$outputDir/multiqc/multiqc_data/fastp_filtered_reads_plot.txt"),
                        path("$outputDir/multiqc/multiqc_data/bclconvert_sample_counts_Counts_per_lane.txt")
                    ).match("multiqc") },
                { assert snapshot(
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Sample1_S1_L001.fastp.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Sample1_S1_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Sample1_S1_L001_summary.txt"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Undetermined_S0_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Reports/").list(),
                    ).match("bclconvert") }
            )
        }

    }

}
