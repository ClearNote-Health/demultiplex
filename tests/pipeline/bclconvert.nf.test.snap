{
    "bclconvert": {
        "content": [
            "Sample1_S1_L001.fastp.fastq.gz:md5,64639ff33d4b07d62a9079f88cbbd373",
            "Sample1_S1_L001_R1_001.fastq.gz:md5,0675fb6365322eaafb33c0f8e862b54b",
            "Sample1_S1_L001_summary.txt:md5,510c6b8a26ff8d530950c05b3ca5a1c6",
            "Undetermined_S0_L001_R1_001.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
            [
                "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                "SampleSheet.csv:md5,ee5db2e12754e069998b0a96e535238c",
                "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                "fastq_list.csv:md5,ae1470cd741e33b87a4ad26f849a50f9"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-07T15:04:02.812091"
    },
    "software_versions": {
        "content": [
            "{BCLCONVERT={bclconvert=4.3.6}, FALCO={falco=1.2.1}, FASTP={fastp=0.23.4}, MD5SUM={md5sum=8.3}, Workflow={nf-core/demultiplex=v1.5.1dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-25T20:31:51.596302017"
    },
    "multiqc": {
        "content": [
            "bclconvert_lane_counts.txt:md5,68de4c4513456af9fd15e90c7145596a",
            "fastp_filtered_reads_plot.txt:md5,fe76dc8c1650e83e16f547c1fe8a0767",
            "bclconvert_sample_counts_Counts_per_lane.txt:md5,f12a2fb8c83cfd6262fcb1ea6dada571"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.4"
        },
        "timestamp": "2024-08-01T14:51:11.574966834"
    }
}
